src/applications/disability-benefits/all-claims/submit-transformer.js

```diff
@@ -0,0 +1,348 @@
+  const claimedConditions = formData.ratedDisabilities
+    ? formData.ratedDisabilities.map(d => d.name.toLowerCase())
+    : [];
+
+  // Depending on where this is called in the transformation flow, we have to use different key names.
+  // This assumes newDisabilities is removed after it's split out into its primary and secondary counterparts.
+  [
+    'newDisabilities',
+    'newPrimaryDisabilities',
+    'newSecondaryDisabilities',
+  ].forEach(key => {
+    if (formData[key]) {
+      // Add new disabilities to claimed conditions list
+      formData[key].forEach(disability =>
+        claimedConditions.push(disability.condition.toLowerCase()),
+      );
+    }
+  });
+  return claimedConditions;
+}
+
+/**
+ * Transforms the related disabilities object into an array of strings. The condition
+ *  name only gets added to the list if the property value is truthy and is in the list
+ *  of conditions claimed on the application.
+ *
+ * @param {Object} object - The object with dynamically generated property names
+ * @param {Object} formData - The whole form data; used to get claimed condition names
+ * @return {Array} - An array of the property names with truthy values
+ *                   NOTE: This will return all lower-cased names
+ */
+export function transformRelatedDisabilities(object, claimedConditions) {
+  return Object.keys(object)
+    .filter(
+      // The property name will be normal-cased in the object, but lower-cased in claimedConditions
+      key => object[key] && claimedConditions.includes(key.toLowerCase()),
+    )
+    .map(key => key.toLowerCase());
+}
+
+/**
+ * Returns an array of the maximum set of PTSD incident form data field names
+ */
+export function getFlatIncidentKeys() {
+  const incidentKeys = [];
+
+  for (let i = 0; i < PTSD_INCIDENT_ITERATION; i++) {
+    incidentKeys.push(`incident${i}`);
+  }
+  for (let i = 0; i < PTSD_INCIDENT_ITERATION; i++) {
+    incidentKeys.push(`secondaryIncident${i}`);
+  }
+
+  return incidentKeys;
+}
+
+/**
+ * Concatenates incident location address object into location string. This will ignore null
+ *  or undefined address fields
+ * @param {Object} incidentLocation location address with city, state, country, and additional details
+ * @returns {String} incident location string
+ */
+export function concatIncidentLocationString(incidentLocation) {
+  return [
+    incidentLocation.city,
+    incidentLocation.state,
+    incidentLocation.country,
+    incidentLocation.additionalDetails,
+  ]
+    .filter(locationField => locationField)
+    .join(', ');
+}
+
+export function getPtsdChangeText(changeFields) {
+  return Object.keys(changeFields)
+    .filter(
+      key =>
+        key !== 'other' &&
+        key !== 'otherExplanation' &&
+        PTSD_CHANGE_LABELS[key],
+    )
+    .map(key => PTSD_CHANGE_LABELS[key]);
+}
+
+export function transform(formConfig, form) {
+  // Define the transformations
+  const filterSelectedRatedDisabilities = formData =>
```

Comment:
This isn't a thing anymore as of my merged PR, but you'll probably find that out when resolving merge conflicts